x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/persistence/DataFrameTransformsConfigManager.java

```diff
@@ -205,6 +213,39 @@ public void getTransformConfiguration(String transformId, ActionListener<DataFra
         }));
     }
 
+    /**
+     * Get the transform configuration for a given transform id. This function is only for internal use. For transforms returned via GET
+     * data_frame/transforms, see the TransportGetDataFrameTransformsAction
+     *
+     * @param transformId the transform id
+     * @param configAndVersionListener listener to call after inner request has returned
+     */
+    public void getTransformConfigurationForUpdate(String transformId,
+                                                   ActionListener<Tuple<DataFrameTransformConfig,
+                                                       SeqNoPrimaryTermPair>> configAndVersionListener) {
```

Comment:
when I suggested the optimistic lock, I was thinking of using the internal `_version` field: https://www.elastic.co/blog/elasticsearch-versioning-support.

Reply:
@hendrikmuhs _version is no longer supported

Reply:
:+1: good to know